docs/mypy.md

```diff
@@ -0,0 +1,30 @@
+# mypy static checker
+
+Zulip now uses mypy. mypy is a compile-time type checker for python.
+For information about what mypy is, visit [http://mypy-lang.org/](http://mypy-lang.org/).
+
+mypy's type checking has been added to Zulip's Travis CI checks.
+
+## Running mypy on Zulip's code
+
+To run mypy on Zulip's code, you'll need to make sure that mypy is installed.
+mypy only runs on python 3. Zulip doesn't use the latest version of mypy available on pypi.
+It uses a specific commit from [mypy's git repository](https://github.com/python/mypy).
+
+*   If you installed Zulip's development environment using vagrant or provision.py,
+    mypy is already installed. To switch to an appropriate python 3 virtual environment, run
+
+        source /srv/zulip-py3-venv/bin/activate
+
+*   If you have not installed Zulip using the above methods, you will have to manually set up
+    a python 3 virtual environment and install mypy in it. Run the following code from Zulip's
+    root directory:
+
+        virtualenv -p python3 ../zulip-py3-venv
+        source ../zulip-py3-venv/bin/activate
+        pip install --upgrade pip
+        pip install -r py3_test_reqs.txt
```

Comment:
Hmm, actually, having now thought about this more, I think we do still want to make a `tools/install-mypy`.  The reason is that for developers following the "by hand" development instructions, they'll need a way to install mypy, and I think it'll be a lot better to just add `tools/install-mypy` to the "All systems" list in those docs than this more complex process than need to add this 4-step installation process there.

I agree that we'll want to eventually have a Python 3 virtualenv installed inside provision.py when we move to Python 3 as a part of the default installation, but in the meantime I'd rather keep the process simple and implemented in just one place.

Does that reasoning make sense to you?  I don't feel strongly on this front so definitely speak up if you disagree!